src/io/fasta.rs

```diff
@@ -1491,4 +1552,59 @@ ATTGTTGTTTTA
         assert!(fs::remove_file(path).is_ok());
         assert_eq!(actual, expected)
     }
+
+    #[test]
+    fn test_reader_from_file_plaintext() {
+        let path = Path::new("test.fa");
+        {
+            let mut file = fs::File::create(path).unwrap();
+            file.write_all(&FASTA_FILE).unwrap();
+        }
+
+        let expected = Reader::new(FASTA_FILE).records().map(|r| r.unwrap());
+
+        let actual = Reader::from_file(path)
+            .unwrap()
+            .records()
+            .map(|r| r.unwrap());
+
+        for (r1, r2) in expected.zip(actual) {
+            assert_eq!(r1.id(), r2.id());
+            assert_eq!(r1.desc(), r2.desc());
+            assert_eq!(r1.seq(), r2.seq());
```

Comment:
Maybe you could just add derived traits `Eq` and `PartialEq` to `Record` and that would make these assertions cleaner?